kubert/src/lease.rs

```diff
@@ -300,7 +296,17 @@ impl LeaseManager {
                 }
 
                 // Update the claim and broadcast it to all receivers.
-                claim = self.ensure_claimed(&claimant, &params).await?;
+                match self.ensure_claimed(&claimant, &params).await {
+                    Ok(new_claim) => claim = new_claim,
+                    Err(error) if Self::is_retryable(&error) => {
+                        tracing::debug!(%error, "failed to claim lease, retrying");
+                        // TODO(eliza): this will immediately retry (since the
+                        // expiration should have elapsed)...should we back off
+                        // here?
+                        continue;
+                    }
+                    Err(error) => return Err(error),
```

Comment:
This is still a problem, though. Right? What happens if there's an I/O error? What happens if the resource doesn't exist? What happens if we don't have sufficient RBAC?

Reply:
Not totally sure if I understand the question you're asking here --- we have two separate match arms, one for errors that we consider retryable, and another for errors that we don't consider retryable. This is the case where we bubble up the error if we don't want to retry it. The logic for determining if we want to retry an error or consider it fatal is here: https://github.com/olix0r/kubert/blob/00577094bfe9b904cc6d15464efa2aab74b31404/kubert/src/lease.rs#L488-L519